watchtower-plugin/src/dbm.rs

```diff
@@ -216,29 +216,39 @@ impl DBM {
         &self,
         tower_id: TowerId,
         user_id: UserId,
-    ) -> Option<RegistrationReceipt> {
-        let mut stmt = self
-            .connection
-            .prepare(
-                "SELECT available_slots, subscription_start, subscription_expiry, signature
-                    FROM registration_receipts 
-                    WHERE tower_id = ?1 AND subscription_expiry = (SELECT MAX(subscription_expiry) 
-                        FROM registration_receipts 
-                        WHERE tower_id = ?1)",
-            )
-            .unwrap();
+        subscription_start: Option<u32>,
+        subscription_expiry: Option<u32>,
+    ) -> Vec<RegistrationReceipt> {
```

Comment:
I think we should keep this an `Option`, so its either `None` or `Some(Vec<...>)`

Reply:
The thing is, after reworking this function to return a vector, it should never return `None`. It will return an empty vector if 1- the tower wasn't found & 2- the given range doesn't intersect with any subscription.